README.md

```diff
@@ -154,6 +154,69 @@ IResult<string> result.Map(value => value.toString());
 
 If `result` represents an failure all calls to `Map` and `FlatMap` are ignored. See `FaunaDB.Types.Result`.
 
+### How to work with user defined classes
+
+Instead of create manually your objects using our dsl (via `Obj()` method) it's possible to use our `Encoder` class and convert any user defined class to an equivalent `Value` type.
+
+Consider the following `Product` class:
+
+```csharp
+class Product
+{
+    [FaunaField("description")]
+    public string Description { get; set; }
+
+    [FaunaField("price")]
+    public double Price { get; set; }
+
+    [FaunaConstructor]
+    public Product(string description, double price)
+    {
+        Description = description;
+        Price = price;
+    }
+}
+```
+
+It's possible to create and persist an instance of `Product` on FaunaDB using:
+
+```csharp
+Product product = new Product("Smartphone", 649.90);
+
+await client.Query(
+    Create(
+        Ref("classes/product"),
+        Obj("data", Encoder.Encode(product))
+    )
+);
+```
+
+Also it's possible to convert back from a `Value` type to a `Product` type using our `Decoder` class:
```

Comment:
I'd probably rephrase, dropping the "it's possible":

"To convert from a `Value` type back to the `Product` type, you can use a `Decoder`:"